satpy/readers/tropomi_l2.py

```diff
@@ -89,9 +90,13 @@ def available_datasets(self, configured_datasets=None):
             matches = self.file_type_matches(ds_info['file_type'])
             # we can confidently say that we can provide this dataset and can
             # provide more info
-            if matches and var_name in self:
+            assembled = var_name in ['assembled_lat_bounds', 'assembled_lon_bounds']
```

Comment:
Don't you need to check also that `latitude_bounds` and `longitude_bounds` are in self ? Otherwise you can't build the assembled version, right ?

Reply:
@mraspaud 
Yes. How about this:
```
            assembled = var_name in ['assembled_lat_bounds', 'assembled_lon_bounds']
            bounds_exist = 'latitude_bounds' in self and 'longitude_bounds' in self
            if (matches and var_name in self) or (assembled and bounds_exist):
```
maybe it's better to put the `bounds_exist` outside the for loop?

Reply:
yes, sounds good.